### PR TITLE
Refactor/remove experimental decorators attribute validate

### PR DESCRIPTION
**Pull request by dsalazar357**
### Description
Removes experimental decorators for `@attributes` and `@validate`
### Changes
- Refactor default attributes definition
- Removes references to `schema structure` in model an store
- Fixes tests to match new default attributes

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -283,3 +283,17 @@ export function deriveIdQueryStrings (ids, restOfUrl = '') {
...
+  if (value instanceof Date) {
+    return 'Date'
+  } else if (Array.isArray(value)) {
+    return 'Array'
+  } else if (value instanceof Object) {
+    return 'Object'
+  } else if (typeof value === 'string') {
+    return 'String'
+  } else return undefined
+}
```
Comment (dudeofawesome) [inline src/utils.js:299]: I think it makes more sense to just do the type checking inline.

File: spec/Model.spec.js [hunk 2]
```diff
@@ -33,7 +40,25 @@ class Note extends Model {
...
+    }
+  }
+}
+
+function validatesString (property) {
+  return {
+    isValid: !isEmptyString(property),
+    errors: [{
+      key: 'blank',
+      message: 'can\'t be blank'
```
Comment (dudeofawesome) [inline spec/Model.spec.js:59]: Prefer to use `"` or \` when making a string with a `'`

**Review by dudeofawesome — APPROVED** [review]

File: spec/Model.spec.js [hunk 3]
```diff
@@ -15,14 +14,22 @@ import {
...
 
-  @attribute(String) description
+  get attributeDefinitions () {
+    return {
+      description: {
+        transformer: toString,
+        defaultValue: ''
+      }
+    }
+  }
```
Comment (jondayton) [inline spec/Model.spec.js:32]: why is this a getter instead of an attribute?

File: src/Model.js [hunk 4]
```diff
@@ -695,14 +706,16 @@ class Model {
...
+        if (dataType === 'Array' || dataType === 'Object') {
           attr = toJS(value)
-        } else if (DataType.name === 'Date') {
+        } else if (dataType === 'Date') {
           attr = makeDate(value).toISOString()
+        } else if (dataType === 'String') {
+          attr = value.toString()
         } else {
-          attr = DataType(value)
+          if (attributeDefinitions[key].transformer) { attr = attributeDefinitions[key].transformer(value) }
```
Comment (jondayton) [inline src/Model.js:718]: should the transformer be first now? previously 'Date', 'Array' were acting as a type of transformer

File: src/Model.js [hunk 5]
```diff
@@ -595,9 +603,16 @@ class Model {
...
    * @return {Object}
    */
+
+ /*
+  *Old attributeDefinitions getter from schema.structure =
   get attributeDefinitions () {
     const { type } = this.constructor
-    return schema.structure[type] || {}
+    return schema.structure[type]
+  } */
```
Comment (jondayton) [inline src/Model.js:612]: delete this code

File: src/Store.js [hunk 6]
```diff
@@ -102,7 +102,7 @@ class Store {
    * @return {Object}
    */
   pickAttributes = (properties, type) => {
-    const attributeNames = Object.keys(this.schema.structure[type])
+    const attributeNames = Object.keys(properties)
```
[the remark this answers is not in the document]
Comment (jondayton) [reply src/Store.js:105]: This is meant to make sure you're only picking properties that are defined as attributes, not other properties on the object.

**Review by jondayton — CHANGES_REQUESTED** [review]

**Review by jondayton — APPROVED** [review]
Looks good! bump to version `5.0.0-rc1`

**Pull request merged**